include/raf/sharding.h

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*!
+ * \file sharding.h
+ * \brief Definition of sharding specifications
+ */
+#pragma once
+#include "./value.h"
+#include <sstream>
+
+namespace raf {
+namespace sharding {
+
+using namespace raf::ir;
+using namespace raf::value;
+
+class BaseShardSpecObj : public ValueObj {
+ public:
+  void VisitAttrs(tvm::AttrVisitor* v) {
+  }
+  static constexpr const uint32_t _type_index = ir::TypeIndex::kDynamic;
+  static constexpr const char* _type_key = "raf.sharding.BaseShardSpec";
+  RAF_BASE_OBJECT(BaseShardSpecObj, ValueObj);
+};
+
+class BaseShardSpec : public Value {
+ public:
+  RAF_OBJECT_REF(BaseShardSpec, Value, BaseShardSpecObj);
+};
+
+class ShardSpecObj final : public BaseShardSpecObj {
+ public:
+  bool mutable_;
```

Comment:
what does this `mutable_` use for? `ShardSpec` is not a mutable object ref.

Reply:
the explanation is written the commit https://github.com/awslabs/raf/pull/115/commits/cff182bd6a3ff99ede15255f8a7977a6d47941b2.